src/ensembl/src/content/app/custom-download/containers/content/result-holder/ResultHolder.tsx

```diff
@@ -0,0 +1,141 @@
+import React, { useEffect } from 'react';
+import { connect } from 'react-redux';
+import { RootState } from 'src/store';
+import {
+  getPreviewResult,
+  getIsLoadingResult
+} from '../../../state/customDownloadSelectors';
+
+import { getAttributes } from '../attributes-accordion/state/attributesAccordionSelector';
+import { getFilters } from '../filter-accordion/state/filterAccordionSelector';
+
+import AttributesSection from 'src/content/app/custom-download/types/Attributes';
+
+import {
+  setPreviewResult,
+  setIsLoadingResult,
+  fetchPreviewResult
+} from '../../../state/customDownloadActions';
+
+import styles from './ResultHolder.scss';
+
+import { CircleLoader } from 'src/shared/loader/Loader';
+
+import {
+  getSelectedAttributes,
+  formatResults,
+  getEndpointUrl
+} from './helpers';
+
+type Props = StateProps & DispatchProps;
+
+const ResultHolder = (props: Props) => {
+  const selectedAttributes: AttributesSection = getSelectedAttributes(
+    props.attributes
+  );
+
+  useEffect(() => {
+    if (!selectedAttributes.length && props.preview.results) {
+      props.clearPreviewResult([]);
+      return;
+    }
+    props.setIsLoadingResult(true);
+
+    const endpointURL = getEndpointUrl(selectedAttributes);
+    if (selectedAttributes.length) {
+      props.fetchPreviewResult(endpointURL);
+    }
+  }, [props.attributes, props.filters]);
+
+  useEffect(() => {
+    props.setIsLoadingResult(false);
+  }, [props.preview]);
+
+  if (props.preview && !props.preview.results && props.isLoadingResult) {
+    return (
+      <>
+        <div className={styles.loaderWrapper}>
+          <CircleLoader />
+        </div>
+        <div className={styles.wrapper}>
+          {Array(15)
+            .fill('')
```

Comment:
Instead of `Array(15).fill()`, `[...Array(15)]`.

what is 15, by the way?

Reply:
I'm trying to display 15 empty result cards on the initial load just so the loader appears nicely.

Reply:
This is really strange (and would probably cause a scroll to appear on a small screen?). Perhaps we could just have a layer with a spinner and overlay it atop the results area while loading results (and the results area meanwhile will either be completely blank or show previous results while waiting for the update to happen)?

Reply:
Updated the loader as discussed.